### DISCUSSION TITLE
SubstructureMatch of an specific part.  Case dNTP molecules.

### DISCUSSION THREAD

**Discussion by davidRFB** [rdkit/rdkit#5180] [Q&A] [answered]
Dear all, 

Hope everything is going well. I have been using the substructureMatch for indentify if a smiles contain a  nucleobases. For example : 
![image](https://user-images.githubusercontent.com/28023319/162484417-73058b3c-04ae-4821-a4f3-9f5805f5e197.png)
contains a modified guanine: 
![image](https://user-images.githubusercontent.com/28023319/162484490-ccac9823-f70e-4ffb-909d-de24d4ad33c6.png)
the method GetSubstructureMatch returns the match
![image](https://user-images.githubusercontent.com/28023319/162484581-7e1fc898-fa56-47e9-aebd-4dbe245c3283.png)

However I need to identify if the nucleobase contains any modification as, other atoms (F) or changes in functional groups. Anyone has an idea of how to take this into account. 

I need a False in this Substructure match. I was thinking in consider hydrogens but the addition to hydrogens to the Nitrogen bonded to the sugar will also affect the result. 

Thank you very much,
David

**Comment by greglandrum** [comment, marked as answer]
@davidRFB This is easiest to do with the function `AdjustQueryProperties()`. Here's a simple example:
```
In [2]: m = Chem.MolFromSmiles("*-C1=NC=CC=N1")

In [3]: q = Chem.MolFromSmiles("*-C1=NC=CC=N1")

In [4]: q = Chem.AdjustQueryProperties(q)

In [5]: Chem.MolToSmarts(q)
Out[5]: '*-[#6&D3]1:[#7&D2]:[#6&D2]:[#6&D2]:[#6&D2]:[#7&D2]:1'

In [6]: Chem.MolFromSmiles("CCCc1ncccn1").HasSubstructMatch(q)
Out[6]: True

In [7]: Chem.MolFromSmiles("CCCc1ncc(C)cn1").HasSubstructMatch(q)
Out[7]: False
```

**Reply by davidRFB** [reply]
@greglandrum  ! Thank you very much for your reply ! 
Unfortunatelly in my case there is no difference but I do not know why. 
I am running the following code. 
```
guanine = Chem.MolFromSmiles('C1=NC2=C(N1)C(=O)NC(=N2)N')
xg4 = Chem.MolFromSmiles('c1nc2c(n1[C@H]3C[C@@H]([C@H](O3)CO[P@@](=O)(N[P@@](=O)(O)OP(=O)(O)O)O)O)N=C(NC2=O)N')
gt8 = Chem.MolFromSmiles('C([C@@H]1[C@H]([C@H]([C@@H](O1)N2C3=C(C(=O)NC(=N3)N)NC2=O)O)O)O[P@@](=O)(O)O[P@@](=O)(O)OP(=O)(O)O')
fz4 = Chem.MolFromSmiles('C1[C@@H]([C@H](O[C@H]1n2c3c(nc2Cl)C(=O)N=C(N3)N)COP(=O)(O)O)O')
mols = [guanine,xg4, gt8, fz4]
```
![image](https://user-images.githubusercontent.com/28023319/162789873-3c6d82b4-249c-4a99-9b10-140a29eb9689.png)

However 
```
guanine = Chem.AdjustQueryProperties(guanine)
for mol in mols[1:]:
    print(mol.HasSubstructMatch(guanine))

>False
>False
>False
```
but should be 
>True
>False
>False


Do you think its for the connection between the base and the sugar in N9.  or the changes in hydrogens around the aromatic structure.
Best,
David

**Reply by greglandrum** [reply]
You need to make sure that you have the correct tautomer of guanine and then add the dummy atom indicating the allowed substitution point. Try this:
```
guanine = Chem.MolFromSmiles("NC1=NC2=C(N=CN2-*)C(=O)N1")
```

**Reply by davidRFB** [reply]
works perfectly ! Thank you very much !
